python/cudf/cudf/tests/dataset_generator.py

```diff
@@ -200,13 +255,173 @@ def generate(
+                            dtype=dtype, size=cardinality
+                        ),
+                        is_sorted=False,
+                        dtype=dtype,
+                    )
+                )
+            elif dtype.kind in ("U", "O"):
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=lambda cardinality=cardinality: [
+                            mimesis.random.random.schoice(
+                                string.printable,
+                                meta.get("max_string_length", 1000),
+                            )
+                            for _ in range(cardinality)
+                        ],
+                        is_sorted=False,
+                        dtype=dtype,
+                    )
+                )
+            elif dtype.kind == "M":
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=datetime_generator(size=cardinality),
+                        is_sorted=False,
+                        dtype=np.dtype(dtype),
+                    )
+                )
+            elif dtype.kind == "m":
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=timedelta_generator(size=cardinality),
+                        is_sorted=False,
+                        dtype=np.dtype(dtype),
+                    )
+                )
+            elif dtype.kind == "b":
+                column_params.append(
+                    ColumnParameters(
+                        cardinality=cardinality,
+                        null_frequency=null_frequency,
+                        generator=boolean_generator(cardinality),
+                        is_sorted=False,
+                        dtype=np.dtype(dtype),
+                    )
+                )
+            else:
+                raise TypeError(f"Unsupported dtype: {dtype}")
+            # TODO: Add List column support once
+            # https://github.com/rapidsai/cudf/pull/6075
+            # is merged.
+
+    df = get_dataframe(
+        Parameters(num_rows=rows, column_parameters=column_params, seed=seed,),
+        use_threads=True,
+    )
+
+    return df
+
+
+def int_generator(dtype, size):
+    iinfo = np.iinfo(dtype)
+    return lambda: np.random.randint(
+        low=iinfo.min, high=iinfo.max, size=size, dtype=dtype,
+    )
+
+
+def float_generator(dtype, size):
+    finfo = np.finfo(dtype)
+    return (
+        lambda: np.random.uniform(
+            low=finfo.min / 2, high=finfo.max / 2, size=size,
+        )
+        * 2
```

Comment:
Why are we multiplying by 2 instead of just generating directly to min / max?

Reply:
This is because current NumPy implementation where if we provide limits of `float` we get overflow error.

```python
>>> import numpy as np
>>> finfo = np.finfo("float64")
>>> np.random.uniform(finfo.min, finfo.max, 10)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "mtrand.pyx", line 959, in numpy.random.mtrand.RandomState.uniform
OverflowError: Range exceeds valid bounds
>>> np.random.uniform(finfo.min, finfo.max/2, 10)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "mtrand.pyx", line 959, in numpy.random.mtrand.RandomState.uniform
OverflowError: Range exceeds valid bounds
>>> np.random.uniform(finfo.min/2, finfo.max/2, 10)
array([ 2.80336379e+307, -1.84988527e+307,  6.00097659e+307,
        1.20549184e+307, -2.56658766e+307,  5.69528438e+307,
        4.78719084e+307,  4.34910856e+306,  7.78469138e+307,
       -8.57599676e+306])
>>> np.random.uniform(finfo.min/2, finfo.max/2, 10) * 2
array([ 1.26865884e+307,  7.70631537e+307, -1.46105042e+308,
        1.44618332e+308,  1.59663611e+308,  7.53097671e+307,
       -8.51266949e+307,  1.78258227e+308, -1.05992116e+307,
        1.74548735e+308])
```